src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,251 @@
+        String[] words = task.split("/by");
+        tasks.add(new Deadline(words[0], words[1]));
+    }
+
+    private static void updateDone(int updateNumber) {
+        tasks.get(updateNumber).markAsDone();
+    }
+
+    private static void updateDelete(int deleteNumber) {
+        tasks.remove(deleteNumber);
+    }
+
+    private static void printList() {
+        if(totalTasks==0) {
+            System.out.println("List is empty.");
+            return;
+        }
+        System.out.println("Here are the tasks in your list:");
+        for(int taskNumber = 0; taskNumber < totalTasks; taskNumber++) {
+            System.out.println(taskNumber + 1 + "." + tasks.get(taskNumber));
+        }
+    }
+
+    private static void printDone(int updateNumber) {
+        System.out.println("Nice! I've marked this task as done: ");
+        System.out.println(tasks.get(updateNumber));
+    }
+
+    private static void printDelete(int deleteNumber) {
+        System.out.println("Noted. I've removed this task: ");
+        System.out.println(tasks.get(deleteNumber));
+        totalTasks -= 1;
+        System.out.println((totalTasks==1)? "Now you have 1 task in the list." : "Now you have " + totalTasks + " tasks in the list.");
+    }
+
+    private static void printUpdate() {
+        System.out.println("Got it. I've added this task: ");
+        System.out.println(tasks.get(totalTasks));
+        totalTasks++;
+        System.out.println((totalTasks==1)? "Now you have 1 task in the list." : "Now you have " + totalTasks + " tasks in the list.");
+    }
+
+    public static void main(String[] args) {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println(logo);
+        loadFile();
+        System.out.println("Hello! I'm Duke.");
+        System.out.println("What can I do for you?");
+        String command;
+        String lowerCaseCommand;
+        do {
+            Scanner in = new Scanner(System.in);
+            command = in.nextLine();
+
+            // remove all unnecessary spaces in input
+            command = command.trim();
+
+            // input on type of task should be case-insensitive
+            lowerCaseCommand = command.toLowerCase();
+
+            try {
+                switch (lowerCaseCommand) {
+                case "done":
+                    throw new EmptyDoneException();
+                case "delete":
+                    throw new EmptyDeleteException();
+                case "todo": case "event": case "deadline":
+                    throw new EmptyTaskException();
+                }
+            } catch (DukeException e) {
+                continue;
+            }
+            if(lowerCaseCommand.equals("list")) {
+                printList();
+            } else if(lowerCaseCommand.startsWith("done")) {
```

Comment:
In reference to my point earlier on factoring your main method further, you can consider extracting line 168 to line 184 as a new method.

Reply:
Similarly, line 185 to 201 can be in a new method as well.